src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.NativeAOT.targets

```diff
@@ -122,16 +110,13 @@ This file contains the NativeAOT-specific MSBuild logic for .NET for Android.
       <ResolvedFileToPublish Include="$(_NdkSysrootDir)libc++_shared.so" RuntimeIdentifier="$(_OriginalRuntimeIdentifier)" />
       <!-- Satellite assemblies -->
       <IlcSatelliteAssembly Include="$(_OuterIntermediateSatelliteAssembliesWithTargetPath)" />
+      <!-- Native support libraries -->
+      <NativeLibrary Include="@(RuntimePackAsset->WithMetadataValue('Filename', 'libnet-android-nativeaot.$(Configuration.ToLower())'))" />
+      <NativeLibrary Include="@(RuntimePackAsset->WithMetadataValue('Filename', 'libxa-shared-bits.$(Configuration.ToLower())'))" />
```

Comment:
The `_AndroidComputeIlcCompileInputs` target runs after `SetupOSSpecificProps` so the items added to `NativeLibrary` here never reach the linker.

Reply:
Also, the libraries are named `libxa-shared-bits-` (dash at the end), not `libxa-shared-bits.`. We should probably fix that to be consistent with `libnet-android-nativeaot`.

Reply:
I solved the `_ZNSt6__ndk16threadD1Ev` issue above by fixing linking of `libc++_shared.so`. The build scripts add it to `ResolvedFileToPublish` but it's stripped later in `_CalculateAndroidFilesToPublish` and never passed to the linker.

That gets us to the next issue with the typemap: `dlopen failed: cannot locate symbol "managed_to_java_map_module_count" referenced by "/data/app/~~FVUgONKNg7TjYrGKp0H5Ow==/com.emclient.mailclient-Ockt9-20CfZzFfra9XWwrw==/lib/x86_64/libMailClient.Mobile.Android.so"`